src/Mvc/Mvc.Core/src/Infrastructure/ProblemDetailsJsonConverter.cs

```
@@ -0,0 +1,142 @@
+            }
+            else if (TryReadStringProperty(ref reader, Title, out propertyValue))
+            {
+                value.Title = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Detail, out propertyValue))
+            {
+                value.Detail = propertyValue;
+            }
+            else if (TryReadStringProperty(ref reader, Instance, out propertyValue))
+            {
+                value.Instance = propertyValue;
+            }
+            else if (reader.ValueTextEquals(Status.EncodedUtf8Bytes))
+            {
+                if (!reader.Read())
+                {
+                    throw new InvalidDataException(Resources.UnexpectedJsonEnd);
+                }
+
+                if (reader.TokenType == JsonTokenType.Null)
+                {
+                    // Nothing to do here.
+                }
+                else if (reader.TokenType == JsonTokenType.Number)
+                {
+                    value.Status = reader.GetInt32();
+                }
+            }
+            else
+            {
+                var key = reader.GetString();
+                reader.Read();
+                value.Extensions[key] = JsonSerializer.Deserialize(ref reader, typeof(object), options);
+            }
+        }
+
+        internal static bool TryReadStringProperty(ref Utf8JsonReader reader, JsonEncodedText propertyName, out string value)
+        {
+            if (!reader.ValueTextEquals(propertyName.EncodedUtf8Bytes))
+            {
+                value = default;
+                return false;
+            }
+
+            if (!reader.Read())
+            {
+                throw new InvalidDataException(Resources.UnexpectedJsonEnd);
+            }
+
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw new InvalidDataException(Resources.FormatUnexpectedJsonToken(reader.TokenType));
+            }
+
+            value = reader.GetString();
```

I don't think any of this is really necessary right? You can just call `GetString()` and let it throw.

---

Yup